Migrate THCTensor_copyIgnoringOverlaps to ATen

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63505
* #63504
* #63429
* #63428

This isn't a public operator, just a helper function used in CUDA_tensor_apply.

Differential Revision: [D30441305](https://our.internmc.facebook.com/intern/diff/D30441305)

peterbell10 committed:
Migrate THCTensor_copyIgnoringOverlaps to ATen

This isn't a public operator, just a helper function used in CUDA_tensor_apply.

[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate THCTensor_copyIgnoringOverlaps to ATen"


This isn't a public operator, just a helper function used in CUDA_tensor_apply.

[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate THCTensor_copyIgnoringOverlaps to ATen"


This isn't a public operator, just a helper function used in CUDA_tensor_apply.

[ghstack-poisoned]

ngimel reviewed (approved):
One day we'll get rid of CUDA_tensor_Apply2, and yeah, we probably should prohibit overlapping writes even before that. 
As far as I can see, it's used only for distributions (poisson, gamma, bernoulli), and those shouldn't accept self-overlapping tensor.

peterbell10 committed:
Update on "Migrate THCTensor_copyIgnoringOverlaps to ATen"


This isn't a public operator, just a helper function used in CUDA_tensor_apply.

[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate THCTensor_copyIgnoringOverlaps to ATen"


This isn't a public operator, just a helper function used in CUDA_tensor_apply.

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30441305).
